docs/src/staking/README.md

```diff
@@ -0,0 +1,66 @@
+# Staking on Solana
+
+## What is Staking?
+Some high level info about PoS networks and staking.
+
+https://lunie.io/guides/what-is-staking/
```

Comment:
I'm not going to include these links, they just had some good info.  Lunie's page felt like a really nice overview, though long for this scope.